core/src/test/java/org/wso2/msf4j/InterceptorTestBase.java

```diff
@@ -1,61 +1,167 @@
 /*
- * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
 package org.wso2.msf4j;
 
-import org.testng.collections.Maps;
+import com.google.gson.Gson;
+import org.apache.commons.io.IOUtils;
+import org.wso2.msf4j.formparam.util.StreamUtil;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.net.HttpURLConnection;
 import java.net.URI;
 import java.net.URL;
-import java.util.Collections;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
 import java.util.Map;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.MediaType;
 
 /**
- * Base class of MSF4J Interceptor tests.
+ * Base class for testing interceptors.
  */
-class InterceptorTestBase {
+abstract class InterceptorTestBase {
+
+    private final Gson gson = new Gson();
+    protected static URI baseURI;
 
-    static URI baseURI;
+    /**
+     * @param path      http path
+     * @param keepAlive should the connection be kept alive
+     * @param headers   request headers
+     * @return object from response
+     * @throws IOException on any exception
+     */
+    protected int doGetAndGetStatusCode(String path, boolean keepAlive, Map<String, String> headers)
+            throws IOException {
+        HttpURLConnection urlConn = request(path, HttpMethod.GET, keepAlive, headers);
+        try {
+            return urlConn.getResponseCode();
+        } finally {
+            urlConn.disconnect();
```

Comment:
Removed urlConn.disconnect() statements. Streams in HttpURLConnection instance are closed where required